src/GafferCycles/IECoreCyclesPreview/Renderer.cpp

```diff
@@ -1102,6 +1104,18 @@ class CyclesAttributes : public IECoreScenePreview::Renderer::AttributesInterfac
 				ShaderAssignPair pair = ShaderAssignPair( light, ccl::array<ccl::Node*>() );
 				pair.second.push_back_slow( m_lightShader->shader() );
 				m_shaderCache->addShaderAssignment( pair );
+
+				bool isMuted = m_muteLight.value_or( false );
+				if( light->get_is_enabled() != !isMuted )
```

Comment:
Looking in the Cycles source, it appears that `set_is_enabled()` goes through `set_if_different()` internally anyway, so there's no real benefit to us avoiding identity edits. As I'm still getting to grips with GafferCycles, every additional line of code sparks "is this necessary, or...?" questions, so I think it'd be really worthwhile keeping this as simple as it can be.

Reply:
Fixed in 2a515fc072bed38621490420662667d98e32d6ff